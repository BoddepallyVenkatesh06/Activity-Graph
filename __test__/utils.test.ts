import express from 'express';
import request from 'supertest';
import { Utilities } from '../src/utils';
import { fakeQueryString, fakeQueryStringRes, options } from './fakeInputs';
import { Handlers } from '../src/handlers';
import { createGraph } from '../src/createChart';

describe('Utilities Test', () => {
    const handlers = new Handlers();
    it('Query Options', () => {
        expect(
            fakeQueryString.map((arg) => {
                const utils = new Utilities(arg);
                return utils.queryOptions();
            })
        ).toEqual(fakeQueryStringRes);
    });

    // Testing express routes
    const fakeServer = () => {
        const app = express();
        app.use(express.urlencoded({ extended: false }));
        return app;
    };

    describe('GET /graph with correct credential', () => {
        test('responding', (done) => {
            const app = fakeServer();
            app.get('/graph', handlers.getGraph);
            request(app)
                .get('/graph?username=ashutosh00710')
                .expect('Content-Type', 'image/svg+xml; charset=utf-8')
                .expect('Cache-Control', 'public, max-age=1800')
                .expect(200, done);
        });
    });

    describe('GET /graph with incorrect credential', () => {
        test('responding', (done) => {
            const app = fakeServer();
            app.get('/graph', handlers.getGraph);
            request(app)
                .get('/graph?username=')
                .expect('Content-Type', 'image/svg+xml; charset=utf-8')
                .expect('Cache-Control', 'no-store, max-age=0')
                .expect(200, done);
        });
    });

    //- Chart Function ([Promise] Inside Graph Cards Class) âœ”
    it('Graph Generation', async () => {
        expect.assertions(1);

        const days = [
            {
                contributionCount: 2,
                date: '1',
            },
            {
                contributionCount: 3,
                date: '2',
            },
            {
                contributionCount: 10,
                date: '3',
            },
            {
                contributionCount: 12,
                date: '4',
            },
            {
                contributionCount: 14,
                date: '5',
            },
        ];
        const graph: Promise<string> = await createGraph('line', options, {
            labels: days.map((day) => day.date),
            series: [{ value: days.map((day) => day.contributionCount) }],
        });
        expect(graph).toMatchSnapshot();
    });
});
