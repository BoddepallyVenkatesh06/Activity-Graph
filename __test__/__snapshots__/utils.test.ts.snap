// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilities Test Graph Generation 1`] = `"<svg xmlns:ct=\\"http://gionkunz.github.com/chartist-js/ct\\" width=\\"10\\" height=\\"5\\" class=\\"ct-chart-line\\"><g class=\\"ct-grids\\"><line x1=\\"50\\" x2=\\"50\\" y1=\\"15\\" y2=\\"16\\" class=\\"ct-grid ct-horizontal\\"></line><line x1=\\"50.2\\" x2=\\"50.2\\" y1=\\"15\\" y2=\\"16\\" class=\\"ct-grid ct-horizontal\\"></line><line x1=\\"50.4\\" x2=\\"50.4\\" y1=\\"15\\" y2=\\"16\\" class=\\"ct-grid ct-horizontal\\"></line><line x1=\\"50.6\\" x2=\\"50.6\\" y1=\\"15\\" y2=\\"16\\" class=\\"ct-grid ct-horizontal\\"></line><line x1=\\"50.8\\" x2=\\"50.8\\" y1=\\"15\\" y2=\\"16\\" class=\\"ct-grid ct-horizontal\\"></line><line y1=\\"16\\" y2=\\"16\\" x1=\\"50\\" x2=\\"51\\" class=\\"ct-grid ct-vertical\\"></line></g><g><g class=\\"ct-series ct-series-a\\"><path d=\\"M50,15.9C50.067,15.883,50.133,15.879,50.2,15.85C50.267,15.821,50.333,15.552,50.4,15.5C50.467,15.448,50.533,15.433,50.6,15.4C50.667,15.367,50.733,15.333,50.8,15.3\\" class=\\"ct-line\\"></path><line x1=\\"50\\" y1=\\"15.9\\" x2=\\"50.01\\" y2=\\"15.9\\" class=\\"ct-point\\" ct:value=\\"2\\"></line><line x1=\\"50.2\\" y1=\\"15.85\\" x2=\\"50.21\\" y2=\\"15.85\\" class=\\"ct-point\\" ct:value=\\"3\\"></line><line x1=\\"50.4\\" y1=\\"15.5\\" x2=\\"50.41\\" y2=\\"15.5\\" class=\\"ct-point\\" ct:value=\\"10\\"></line><line x1=\\"50.6\\" y1=\\"15.4\\" x2=\\"50.61\\" y2=\\"15.4\\" class=\\"ct-point\\" ct:value=\\"12\\"></line><line x1=\\"50.8\\" y1=\\"15.3\\" x2=\\"50.809999999999995\\" y2=\\"15.3\\" class=\\"ct-point\\" ct:value=\\"14\\"></line></g></g><g class=\\"ct-labels\\"><text x=\\"50\\" y=\\"36\\" width=\\"0.2\\" height=\\"20\\" class=\\"ct-label ct-horizontal ct-end\\">1</text><text x=\\"50.2\\" y=\\"36\\" width=\\"0.2\\" height=\\"20\\" class=\\"ct-label ct-horizontal ct-end\\">2</text><text x=\\"50.4\\" y=\\"36\\" width=\\"0.20000000000000007\\" height=\\"20\\" class=\\"ct-label ct-horizontal ct-end\\">3</text><text x=\\"50.6\\" y=\\"36\\" width=\\"0.19999999999999996\\" height=\\"20\\" class=\\"ct-label ct-horizontal ct-end\\">4</text><text x=\\"50.8\\" y=\\"36\\" width=\\"30\\" height=\\"20\\" class=\\"ct-label ct-horizontal ct-end\\">5</text><text y=\\"16\\" x=\\"40\\" height=\\"30\\" width=\\"30\\" class=\\"ct-label ct-vertical ct-start\\">0</text></g></svg>"`;
